### DISCUSSION TITLE
How to get several QR-Codes from .PNG

### DISCUSSION THREAD

**Discussion by TerraD** [micjahn/ZXing.Net#564] [Q&A]
I use this library with then [VBA (COM Interop)](https://github.com/micjahn/ZXing.Net/wiki/Using-ZXing.Net-with-VBA-(COM-Interop)) wrapper to extract and read QR-Codes from .PDF. For this I first convert the .PDF page by page to a .PNG. This works pretty well!

```
  Set reader = New BarcodeReader
  reader.options.PossibleFormats.Add BarcodeFormat_QR_CODE
  Set res = reader.DecodeImageFile(strFileToDecode) 
```

However now I stumbled over a .PNG (=page in a PDF-file) that contains 2 QR-Codes - and I need the second one, while the library returns the first. Is there any way to read the content of the second (and further...) QR-codes?

Once I have read the barcode I will be able to discriminate whether it was the correct one.

**Comment by TerraD** [comment]
As a workaround I implemented the following temporary solution: If no qualifying QR-Code is found on the page I rotate the .PNG by 90° and read again. If I still do not find a qualifying QR-Code, I rotate  the .PNG again by 90°. 

Rather primitive, but I think it will help for 99.8% of my application (reading Swiss QR-paymentslips)... - if there is an additional QR-Code somewhere on the bill, it is not very likely, that there is a second one all down.... 

Just realized: Maybe it would find the QR-Code faster and more reliable by rotating by 180°, then 90° and again by 180°...

**Comment by micjahn** [comment]
You can try the method "DecodeImageFileMultiple" to get more than one result if there is more than one code found in the picture.
